### ISSUE TITLE
calendar: potential XSS when using user-supplied timeOnlyTitle translation

### ISSUE THREAD

**Issue by cnsgithub** [primefaces/primefaces#4385]
## 1) Environment
- PrimeFaces version: 6.3-snapshot

## 2) Expected behavior
`timeOnlyTitle `is contained in `PrimeFaces.locales` (https://github.com/primefaces/primefaces/wiki/Locales) and therefore must be considered as user-supplied input.

## 3) Actual behavior
The value is passed through to `jquery.ui.timepicker.js` without being escaped there: 					https://github.com/primefaces/primefaces/blob/f6f316d851da276824fdf70495c160b071386d37/src/main/resources/META-INF/resources/primefaces/jquery/jquery.ui.timepicker.js#L440

This is actually a bug in `jquery.ui.timepicker.js`. However, since PrimeFaces bundles it, it should be fixed here as well.

Basically all strings contained in `PrimeFaces.locales` should be paid extra attention.

**Comment by tandraschko** [comment]
I think we pass "o" to the timepicker somehow?
It would be great if we could do something like this in PF, instead of changes the timepicker.js:
o.timeOnlyTitle = escape(o.timeOnlyTitle);

**Comment by cnsgithub** [comment]
That's kind of problematic, since we cannot know where `o.timeOnlyTitle` is used (in fact we currently know, but things might change :wink: ) and therefore don't know the context it's used in. So we should not assume it's being used in HTML context only.

There is no link to the lib provided in `jquery.ui.timepicker.js` that might be used to check for newer versions?

**Comment by melloware** [comment]
@cnsgithub the TimePicker jquery addon is found here: https://github.com/trentrichardson/jQuery-Timepicker-Addon

**Comment by melloware** [comment]
It looks like the original Author Trent Richardson is no longer maintaining it. https://github.com/trentrichardson/jQuery-Timepicker-Addon/issues/955   We usually make fixes to this component using the pfextensions.js file but since its abandoned we can probably just start editing this add on directly?

**Comment by cnsgithub** [comment]
@melloware I agree.

**Comment by cnsgithub** [comment]
PR: https://github.com/primefaces/primefaces/pull/4426
